default/drive_options.py

```diff
@@ -0,0 +1,49 @@
+from __future__ import print_function
+from django.http import HttpResponse
+from google.auth.transport.requests import Request
+from google_auth_oauthlib.flow import InstalledAppFlow
+from googleapiclient.discovery import build
+from googleapiclient.errors import UnknownFileType
+from googleapiclient.http import MediaFileUpload
+from oauthlib.oauth2 import AccessDeniedError
+from ocdstoucan.settings import OCDS_TOUCAN_CREDENTIALS_DRIVE
+
+
+def upload_to_drive(filename, filepath, format=None):
+    try:
+        credentials = None
+        SCOPES = 'https://www.googleapis.com/auth/drive.file'
+        if not credentials or not credentials.valid:
+            if credentials and credentials.expired and credentials.refresh_token:
+                credentials.refresh(Request())
+            else:
+                flow = InstalledAppFlow.from_client_secrets_file(
+                    OCDS_TOUCAN_CREDENTIALS_DRIVE, SCOPES)
+                credentials = flow.run_local_server(port=0)
+        service = build('drive', 'v3', credentials=credentials)
+
+    except AccessDeniedError:
+        return HttpResponse('Access Denied<br>Acceso Denegado', status=400)
```

Comment:
Here it's better to return a JSON response that could be handled in the client side.

Reply:
Still return `HttpResponse` but handled in the client side with `jqXHR.responseText` when the function fails.